src/mavoscript.js

```diff
@@ -29,11 +29,19 @@ var _ = Mavo.Script = {
 			if (Array.isArray(a)) {
 				result = [];
 				var max = Math.max(a.length, b.length);
+				var leftIdentity = o.leftIdentity === undefined ? o.identity : o.leftIdentity;
+				var rightIdentity = o.rightIdentity === undefined ? o.identity : o.rightIdentity;
+
 				for (let i = 0; i < max; i++) {
-					result[i] = o.scalar(
-						a[i] === undefined ? o.identity : a[i],
-						b[i] === undefined ? o.identity : b[i]
-					);
+					if (a[i] !== undefined && b[i] !== undefined) {
+						result[i] = o.scalar(a[i], b[i]);
+					}
+					else if (a[i] === undefined) {
+						result[i] = o.leftScalar === undefined ? o.scalar(leftIdentity, b[i]) : o.leftScalar(b[i]);
```

Comment:
Since `o.leftScalar` is a function, just checking for truthiness works (i.e. `o.leftScalar ?` instead of `o.leftScalar === undefined ?`). For identities we explicitly compare with undefined, because 0 and "" are falsy.